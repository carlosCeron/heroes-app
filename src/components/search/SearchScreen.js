import React, { useMemo } from "react";
import queryString from 'query-string';
import PropTypes from "prop-types";
import { heroes } from "../../data/heroes";
import HeroCard from "../heroes/HeroCard";
import { useForm } from "../../hooks/useForm";
import { useLocation } from "react-router";
import { getHeroesByName } from "../selectores/getHeroesByName";

const SearchScreen = ({history}) => {

  const location = useLocation();
  const {q = ''} = queryString.parse(location.search);


  const [formValues, handleInputChange] = useForm({
    findElement: q
  });

  const {findElement} = formValues;

  const heroesFilter =   useMemo(() => getHeroesByName(q), [q]);


  const handleSearch = (e) => {
    e.preventDefault();

    console.log(findElement);
    history.push(`?q=${findElement}`);
  }

  return (
    <div>
      <h1>Search Screen</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Search Form</h4>
          <hr />
          <form onSubmit={handleSearch}>
            <input 
            type="text" 
            name="findElement" 
            placeholder="Find your hero" 
            className="form-control"
            value={findElement}
            onChange={handleInputChange} />
            <button type="submit" className="btn m-1  btn-block btn-outline-primary">
              Search..
            </button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />
          {
            heroesFilter.map(hero => (
              <HeroCard key={hero.id} {...hero} />
            ))
          }
        </div>
      </div>
    </div>
  );
};

SearchScreen.propTypes = {};

export default SearchScreen;
