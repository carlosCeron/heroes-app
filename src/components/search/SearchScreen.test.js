import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Route } from "react-router-dom";
import SearchScreen from "./SearchScreen";

describe("probando el componente SearchScreen", () => {
  test("debe de mostrarse correctamente con valores por defecto", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();

  });


  test("debe de mostrarse correctamente con el valor dado", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find('.card-title').text()).toBe('Batman');

  });

});
