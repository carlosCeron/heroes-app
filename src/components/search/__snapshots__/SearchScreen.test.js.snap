// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando el componente SearchScreen debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="findElement"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1  btn-block btn-outline-primary"
          type="submit"
        >
          Search..
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
    </div>
  </div>
</div>
`;
